src/emulator/eventarcEmulator.ts

```diff
@@ -0,0 +1,90 @@
+import * as _ from "lodash";
+import * as express from "express";
+import * as cors from "cors";
+import { logger } from "../logger";
+import { Constants } from "./constants";
+import { EmulatorInfo, EmulatorInstance, Emulators } from "./types";
+import { createDestroyer } from "../utils";
+import { EmulatorLogger } from "./emulatorLogger";
+
+interface RequestWithRawBody extends express.Request {
+  rawBody: Buffer;
+}
+
+export interface EventarcEmulatorArgs {
+  port?: number;
+  host?: string;
+}
+
+export class EventarcEmulator implements EmulatorInstance {
+  private destroyServer?: () => Promise<void>;
+
+  private logger = EmulatorLogger.forEmulator(Emulators.EVENTARC);
+
+  constructor(private args: EventarcEmulatorArgs) {}
+
+  createHubServer(): express.Application {
+    const hub = express();
+
+    hub.use(express.json())
+
+    const helloWorldRoute = `/hello_world`;
+    const publishEventsRoute = `/v1/projects/:project_id/locations/:location/channels/:channel::publishEvents`;
```

Comment:
Double colon intentional? Do we need to escape the character here?

Reply:
Yep, it's intentional. Otherwise it gets picked up as a variable.